service/src/test/java/com/commercetools/pspadapter/payone/IntegrationServiceTest.java

```diff
@@ -0,0 +1,155 @@
+                createTenantFactory(TENANTNAME2, false));
+        integrationService = new IntegrationService(serviceConfig, tenantFactories);
+        integrationService.start();
+
+        HealthResponse result = request();
+        assertThat(result.status).isEqualTo(SUCCESS_STATUS);
+        assertThat(result.body).isEqualTo(expectedBody(SUCCESS_STATUS, SUCCESS_STATUS, SUCCESS_STATUS));
+ }
+
+    @Test
+    public void createHealthResponse_allTenantsAreNotWorking_shouldReturnErrorStatus() {
+        List<TenantFactory> tenantFactories = of(createTenantFactory(TENANTNAME1, true),
+                createTenantFactory(TENANTNAME2, true));
+        integrationService = new IntegrationService(serviceConfig, tenantFactories);
+        integrationService.start();
+
+        HealthResponse result = request();
+        assertThat(result.status).isEqualTo(ERROR_STATUS);
+        assertThat(result.body).isEqualTo(expectedBody(ERROR_STATUS, ERROR_STATUS, ERROR_STATUS));
+
+    }
+
+    @Test
+    public void createHealthResponse_oneTenantIsNotWorking_shouldReturnErrorStatus() {
+        List<TenantFactory> tenantFactories = of(createTenantFactory(TENANTNAME1, true),
+                createTenantFactory(TENANTNAME2, false));
+        integrationService = new IntegrationService(serviceConfig, tenantFactories);
+        integrationService.start();
+
+        HealthResponse result = request();
+        assertThat(result.status).isEqualTo(ERROR_STATUS);
+        assertThat(result.body).isEqualTo(expectedBody(ERROR_STATUS, ERROR_STATUS, SUCCESS_STATUS));
+
+    }
+
+    private TenantFactory createTenantFactory(String tenantName, boolean erroneous) {
+        TenantFactory tenantFactory = Mockito.mock(TenantFactory.class);
+        BlockingSphereClient client = Mockito.mock(BlockingSphereClient.class);
+        if (erroneous) {
+            when(client.execute(Mockito.anyObject())).thenReturn(exceptionallyCompletedFuture(new SphereTimeoutException(new TimeoutException())));
+        } else {
+            PagedQueryResult mockResult = Mockito.mock(PagedQueryResult.class);
+            when(client.execute(Mockito.anyObject())).thenReturn(completedFuture(mockResult));
+        }
+        when(tenantFactory.getBlockingSphereClient()).thenReturn(client);
+        when(tenantFactory.getTenantName()).thenReturn(tenantName);
+        return tenantFactory;
+    }
+
+    private String expectedBody(int globalStatus, int tenant1Status, int tenant2Status) {
+        return "{\"status\":" + globalStatus + ",\"tenants\":{\"tenant2\":" + tenant2Status + ",\"tenant1\":" + tenant1Status + "}," +
+                "\"applicationInfo\":{\"version\":\"1.0\",\"title\":\"applicationTitle\"}}";
+    }
+
+    /**
+     * This assures that the sparkserver has successfully stopped, before the next test is executed.
+     * For more info, please refer: https://github.com/perwendel/spark/issues/731
+     */
+    private void waitUntilServerStops() {
```

Comment:
https://static.javadoc.io/com.sparkjava/spark-core/2.8.0/spark/Service.html#awaitStop--
Just try to update the spark version

Reply:
.83b77bed9